packages/core/src/machines/index.ts

```diff
@@ -635,10 +635,10 @@ export const createAuthMachine = ({
           }),
 
         autoSignIn: async () => {
```

Comment:
I'd try to make an early exit here (the error message is just a sample):
```js
if (typeof window === undefined || !window.location) {
  throw new Error('window is undefined or location does not exist')
}
```

Reply:
it doesn't change the existing logic I think:
!(a && b) <=> (!a || !b)